quil-rs/src/instruction/gate.rs

```diff
@@ -0,0 +1,222 @@
+use std::fmt;
```

Comment:
Like in `quil-py`, I started breaking out instructions into multiple files. I started with Gate and it's counterparts, since I chose it as the pilot instruction for backing a `pyQuil` `AbstractInstruction`. As we implement more instruction constructors, I plan to continue breaking these out, such that the file structure of `quil-rs` and `quil-py` mirror each other.

Reply:
Added a CONTRIBUTING.md